internal/server/middleware.go

```diff
@@ -110,6 +111,25 @@ func authenticateRequest(c *gin.Context, route routeSettings, srv *Server) (acce
 		if err := handleInfraDestinationHeader(rCtx, c.Request.Header); err != nil {
 			return authned, err
 		}
+		if route.idpSync {
+			if route.authenticationOptional {
+				// this should be caught during development
+				return authned, fmt.Errorf("idp sync requires authentication")
+			}
+			// sync the identity info here to keep the UI session in sync with IDP session validity
+			if err := srv.syncIdentityInfo(context.Background(), tx, authned.User, authned.AccessKey.ProviderID); err != nil {
```

Comment:
We don't want to do this sync (external API call to an IDP) with an open transaction that may be holding database row locks.

I think we'll need to adjust the transactions so that we commit it before doing this step.